samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -69,16 +69,19 @@
 
   private final StreamGraphBuilder graphBuilder;
 
+  private final ApplicationRunner runner;
+
   private ContextManager contextManager;
 
-  public StreamOperatorTask(StreamGraphBuilder graphBuilder) {
+  public StreamOperatorTask(StreamGraphBuilder graphBuilder, ApplicationRunner runner) {
     this.graphBuilder = graphBuilder;
+    this.runner = runner;
   }
 
   @Override
   public final void init(Config config, TaskContext context) throws Exception {
     // create the MessageStreamsImpl object and initialize app-specific logic DAG within the task
-    StreamGraphImpl streams = new StreamGraphImpl();
+    StreamGraphImpl streams = new StreamGraphImpl(this.runner);
```

Comment:
To address your comment below, I need to pass in the config now for the job name and id. So I still need to keep streamgraph created here.

Renamed it to streamGraph.